newsfragments/3287.added.md

```diff
@@ -0,0 +1 @@
+`#[new]` methods may now return `Py<Self>` in order to return existing instances
```

Comment:
I think this change requires adjustments to the guide as well.

Reply:
Good point, if the code (or at least, the public API) looks good I can update the guide, wanted to hold off on there until we knew how the API/implementation would impact what needed to be documented.

Reply:
Ok, added an explanation. Let me know if you think it needs more.